src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.google.common;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.common.ReferencePluginConfig;
+
+/**
+ * Base Google Drive batch config. Contains common configuration properties and methods.
+ */
+public abstract class GoogleDriveBaseConfig extends ReferencePluginConfig {
+  public static final String APP_ID = "appId";
+  public static final String CLIENT_ID = "clientId";
+  public static final String CLIENT_SECRET = "clientSecret";
+  public static final String REFRESH_TOKEN = "refreshToken";
+  public static final String ACCESS_TOKEN = "accessToken";
+  public static final String DIRECTORY_IDENTIFIER = "directoryIdentifier";
+
+  public static final String BODY_FIELD_NAME = "body";
+  public static final String OFFSET_FIELD_NAME = "offset";
+
+  private static final String IS_VALID_FAILURE_MESSAGE_PATTERN = "%s has invalid value %s";
+  private static final String IS_SET_FAILURE_MESSAGE_PATTERN = "%s is empty or macro is not available";
+  private static final String CHECK_CORRECTIVE_MESSAGE_PATTERN = "Enter valid %s";
+
+  @Name(APP_ID)
+  @Description("Oauth2 app id")
+  @Macro
+  protected String appId;
+
+  // TODO remove these properties after OAuth2 will be provided by cdap
+  // start of workaround
+  @Name(CLIENT_ID)
+  @Description("OAuth2 client id.")
+  @Macro
+  protected String clientId;
+
+  @Name(CLIENT_SECRET)
+  @Description("OAuth2 client secret.")
+  @Macro
+  protected String clientSecret;
+
+  @Name(REFRESH_TOKEN)
+  @Description("OAuth2 refresh token.")
+  @Macro
+  protected String refreshToken;
+  // end of workaround
+
+  @Name(ACCESS_TOKEN)
+  @Description("OAuth2 access token.")
+  @Macro
+  protected String accessToken;
+
+  @Name(DIRECTORY_IDENTIFIER)
+  @Description("ID of target directory, the last part of the URL.")
+  @Macro
+  protected String directoryIdentifier;
+
+  public GoogleDriveBaseConfig(String referenceName) {
+    super(referenceName);
+  }
+
+  public void validate(FailureCollector collector) {
```

Comment:
We will need to validate
- if the fields with field names are in schema.
- if their type in schema is correct.
- Also please check if we can validate if authentication can be done using credentials given.
- And if possible to check if directoryIdentifier is present.